test/data/stages/bootupd.gen-metadata/diff.json

```diff
@@ -0,0 +1,35 @@
+{
+  "added_files": [
+    "/usr/lib/bootupd",
+    "/usr/lib/bootupd/updates",
+    "/usr/lib/bootupd/updates/BIOS.json",
+    "/usr/lib/bootupd/updates/EFI",
+    "/usr/lib/bootupd/updates/EFI.json",
+    "/usr/lib/bootupd/updates/EFI/BOOT",
+    "/usr/lib/bootupd/updates/EFI/BOOT/BOOTX64.EFI",
+    "/usr/lib/bootupd/updates/EFI/BOOT/fbx64.efi",
+    "/usr/lib/bootupd/updates/EFI/fedora",
+    "/usr/lib/bootupd/updates/EFI/fedora/BOOTX64.CSV",
+    "/usr/lib/bootupd/updates/EFI/fedora/grubx64.efi",
+    "/usr/lib/bootupd/updates/EFI/fedora/mmx64.efi",
+    "/usr/lib/bootupd/updates/EFI/fedora/shim.efi",
+    "/usr/lib/bootupd/updates/EFI/fedora/shimx64.efi"
+  ],
+  "deleted_files": [
+    "/usr/lib/ostree-boot/efi/EFI",
```

Comment:
I'm sure this is all fine but I'm slightly confused why the `generate-update-metadata` command will produce deleted_files? Looking at https://github.com/coreos/bootupd/blob/main/README-devel.md#generating-an-update-payload it does not indicate that this happens. Or am I misunderstanding something here?

Reply:
It moves those files and generates two JSON files for metadata.  Compare the deleted files with the added ones.

Reply:
I believe
```
puts them along with their component files in /usr/lib/bootupd/updates
```
refers to this move.

Reply:
The history is:

- fedora derivatives have dnf/rpm directly write to `/boot/efi`
- ostree is created, strictly separates userspace/kernel updates from bootloader updates and for lack of a better alternative puts the RPM content in `/usr/lib/ostree-boot` where is is inert
- bootupd is created, takes over stuff from `/usr/lib/ostree-boot`

It would clearly make more sense ultimately for rpm-ostree to know that bootupd is in use and handle this more directly of course.

Reply:
Thanks! So AIUI the main reason for the moving is history not a deeper technical reason. That's fine of course, I will approve and if that ever change we just update the test.

Reply:
Resolving this.